otherlibs/xdg/xdg.ml

```diff
@@ -10,28 +10,37 @@ type t =
 
 let ( / ) = Filename.concat
 
-let make t env_var unix_default win32_default =
-  let default = if t.win32 then win32_default else unix_default in
+let make t env_var unix_default win32_var =
+  let default =
+    if t.win32 then
+      match win32_var t.env with
+      | None -> ""
+      | Some s -> s
+    else unix_default
+  in
   match t.env env_var with
   | None -> default
   | Some s when Filename.is_relative s -> default
   | Some s -> s
 
+external get_user_cache_dir : unit -> string = "xdg__get_user_cache_dir"
+
 let cache_dir t =
   let home = t.home_dir in
-  make t "XDG_CACHE_HOME" (home / ".cache") (home / "Local Settings" / "Cache")
+  make t "XDG_CACHE_HOME" (home / ".cache") (fun env ->
+      match get_user_cache_dir () with
```

Comment:
Isn't it easier to just make `get_user_cache_dir` return an option directly?

Reply:
Yes, fixed.